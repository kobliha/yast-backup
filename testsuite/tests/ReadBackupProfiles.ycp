// Backup module - testsuite
//
// ReadBackupProfiles function test
//
// testedfiles: Backup.ycp
//
// $Id$
//

{

import "Testsuite";

map readmap = $[ "target" : $["string" : "", "size" : -1, "tmpdir" : "/tmp" ],
    "proc" : $["cpuinfo" : $[ "value" : $["0" : $[ "flags" : ""]]]],
    "probe" : $["has_smp" : false, "is_uml" : false, "cdrom" : $["manual" : []]],
    ];


map execmap = $[ "target" : $["bash_output" : $["exit":0, "stderr":"", "stdout":"charmap=\"ISO-8859-1\"\n"]]];

Testsuite::Init([ readmap, $[], execmap ] ,nil);

import "Pkg";   // override packamanager

import "Backup";

map READ =
$[ "target" : $[ "ycp" :
$[
  "ahoj" : $[
    `archive_name : "/tmp/ahoj.tar",
    `archive_type : `tgz,
    `backup_all_ext2 : false,
    `backup_none_ext2 : true,
    `backup_pt : true,
    `backup_selected_ext2 : false,
    `default_dir : [
      "/tmp",
      "/var/lock",
      "/var/run",
      "/var/tmp",
      "/var/cache"
    ],
    `description : "",
    `detected_ext2 : nil,
    `detected_fs : nil,
    `dir_list : nil,
    `display : true,
    `do_md5_test : false,
    `ext2_backup : [
    ],
    `fs_exclude : [
      "iso9660"
    ],
    `multi_volume : false,
    `search : false,
    `system : false,
    `unselected_files : [
      "/etc/inittab"
    ],
    `user_volume_size : "",
    `user_volume_unit : nil,
    `volume_size : `fd144
  ],
  "final" : $[
    `archive_name : "/tmp/ahoj.tar",
    `archive_type : `tgz,
    `backup_all_ext2 : false,
    `backup_none_ext2 : true,
    `backup_pt : true,
    `backup_selected_ext2 : false,
    `default_dir : [
      "/tmp",
      "/var/lock",
      "/var/run",
      "/var/tmp",
      "/var/cache"
    ],
    `description : "",
    `detected_ext2 : nil,
    `detected_fs : [
      "devpts",
      "nfs",
      "proc",
      "reiserfs",
      "rootfs",
      "shm"
    ],
    `dir_list : [
      `item (`id ("/tmp"), "/tmp"),
      `item (`id ("/var/lock"), "/var/lock"),
      `item (`id ("/var/run"), "/var/run"),
      `item (`id ("/var/tmp"), "/var/tmp"),
      `item (`id ("/var/cache"), "/var/cache"),
      `item (`id ("/usr"), "/usr"),
      `item (`id ("/home"), "/home"),
      `item (`id ("/local"), "/local"),
      `item (`id ("/etc"), "/etc"),
      `item (`id ("/var"), "/var"),
      `item (`id ("/opt"), "/opt")
    ],
    `display : false,
    `do_md5_test : true,
    `ext2_backup : [
    ],
    `fs_exclude : [
      "bdev",
      "devpts",
      "futexfs",
      "iso9660",
      "nfs",
      "pipefs",
      "proc",
      "ramfs",
      "rootfs",
      "shm",
      "sockfs",
      "tmpfs",
      "usbfs"
    ],
    `multi_volume : false,
    `search : true,
    `system : false,
    `unselected_files : nil,
    `user_volume_size : "",
    `user_volume_unit : `B,
    `volume_size : `fd144
  ]
] ] ];

// read profiles
Testsuite::Test(``(Backup::ReadBackupProfiles()), [READ, $[], execmap], nil);

// dump the result
Testsuite::Dump( Backup::backup_profiles);

}

