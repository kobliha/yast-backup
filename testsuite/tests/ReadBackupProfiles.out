Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir nil
Read	.target.size nil -1
Dir	.autoinst.menuentry.s: []
Read	.target.ycp "/var/adm/YaST/backup/profiles" $["ahoj":$[`archive_name:"/tmp/ahoj.tar", `archive_type:`tgz, `backup_all_ext2:false, `backup_none_ext2:true, `backup_pt:true, `backup_selected_ext2:false, `default_dir:["/tmp", "/var/lock", "/var/run", "/var/tmp", "/var/cache"], `description:"", `detected_ext2:nil, `detected_fs:nil, `dir_list:nil, `display:true, `do_md5_test:false, `ext2_backup:[], `fs_exclude:["iso9660"], `multi_volume:false, `search:false, `system:false, `unselected_files:["/etc/inittab"], `user_volume_size:"", `user_volume_unit:nil, `volume_size:`fd144], "final":$[`archive_name:"/tmp/ahoj.tar", `archive_type:`tgz, `backup_all_ext2:false, `backup_none_ext2:true, `backup_pt:true, `backup_selected_ext2:false, `default_dir:["/tmp", "/var/lock", "/var/run", "/var/tmp", "/var/cache"], `description:"", `detected_ext2:nil, `detected_fs:["devpts", "nfs", "proc", "reiserfs", "rootfs", "shm"], `dir_list:[`item (`id ("/tmp"), "/tmp"), `item (`id ("/var/lock"), "/var/lock"), `item (`id ("/var/run"), "/var/run"), `item (`id ("/var/tmp"), "/var/tmp"), `item (`id ("/var/cache"), "/var/cache"), `item (`id ("/usr"), "/usr"), `item (`id ("/home"), "/home"), `item (`id ("/local"), "/local"), `item (`id ("/etc"), "/etc"), `item (`id ("/var"), "/var"), `item (`id ("/opt"), "/opt")], `display:false, `do_md5_test:true, `ext2_backup:[], `fs_exclude:["bdev", "devpts", "futexfs", "iso9660", "nfs", "pipefs", "proc", "ramfs", "rootfs", "shm", "sockfs", "tmpfs", "usbdevfs", "usbfs"], `multi_volume:false, `search:true, `system:false, `unselected_files:nil, `user_volume_size:"", `user_volume_unit:`B, `volume_size:`fd144]]
Read	.target.dir "/etc/cron.d" nil
Return	true
Dump	$["ahoj":$[`archive_name:"/tmp/ahoj.tar", `archive_type:`tgz, `backup_all_ext2:false, `backup_none_ext2:true, `backup_pt:true, `backup_selected_ext2:false, `cron_settings:$["cron_changed":false, "cronfile":""], `default_dir:["/tmp", "/var/lock", "/var/run", "/var/tmp", "/var/cache"], `description:"", `detected_ext2:nil, `detected_fs:nil, `dir_list:nil, `display:true, `do_md5_test:false, `ext2_backup:[], `fs_exclude:["iso9660"], `multi_volume:false, `search:false, `system:false, `unselected_files:["/etc/inittab"], `user_volume_size:"", `user_volume_unit:nil, `volume_size:`fd144], "final":$[`archive_name:"/tmp/ahoj.tar", `archive_type:`tgz, `backup_all_ext2:false, `backup_none_ext2:true, `backup_pt:true, `backup_selected_ext2:false, `cron_settings:$["cron_changed":false, "cronfile":""], `default_dir:["/tmp", "/var/lock", "/var/run", "/var/tmp", "/var/cache"], `description:"", `detected_ext2:nil, `detected_fs:["devpts", "nfs", "proc", "reiserfs", "rootfs", "shm"], `dir_list:[`item (`id ("/tmp"), "/tmp"), `item (`id ("/var/lock"), "/var/lock"), `item (`id ("/var/run"), "/var/run"), `item (`id ("/var/tmp"), "/var/tmp"), `item (`id ("/var/cache"), "/var/cache"), `item (`id ("/usr"), "/usr"), `item (`id ("/home"), "/home"), `item (`id ("/local"), "/local"), `item (`id ("/etc"), "/etc"), `item (`id ("/var"), "/var"), `item (`id ("/opt"), "/opt")], `display:false, `do_md5_test:true, `ext2_backup:[], `fs_exclude:["bdev", "devpts", "futexfs", "iso9660", "nfs", "pipefs", "proc", "ramfs", "rootfs", "shm", "sockfs", "tmpfs", "usbdevfs", "usbfs"], `multi_volume:false, `search:true, `system:false, `unselected_files:nil, `user_volume_size:"", `user_volume_unit:`B, `volume_size:`fd144]]
